src/timetable/timetable.tsx

```diff
@@ -324,71 +329,78 @@ export const TimeTable = ({
-                        onRightArrowClick={onNextDayClick}
-                        onRefresh={onRefresh}
-                    />
-                </EmptyTableRowHeader>
-                {!loading ? (
-                    <NoResultsFound
-                        type="no-item-found"
-                        illustrationScheme={emptyContent?.illustrationScheme}
-                        description={emptyContent?.description}
-                    />
-                ) : (
-                    <Loader $isEmptyContent={isEmptyContent}></Loader>
-                )}
-            </EmptyTableContainer>
-        );
-    }
-
-    return (
-        <TimeTableContainer
-            ref={tableContainerRef}
-            data-testid={testId}
-            {...otherProps}
-            $loading={loading}
-            $allRecordsLoaded={allRecordsLoaded || !onPage}
-        >
-            <RowColumnHeader
-                $isScrolledY={isScrolledY}
-                $isScrolledX={isScrolledX}
-            >
-                <TimeTableNavigator
+            <Container {...otherProps}>
+                <TimeTableHeader
                     selectedDate={date}
                     loading={loading || loadMore}
                     tableContainerRef={tableContainerRef}
                     minDate={minDate}
                     maxDate={maxDate}
                     totalRecords={totalRecords}
-                    onLeftArrowClick={onPreviousDayClick}
-                    onRightArrowClick={onNextDayClick}
+                    showCurrentDateAsToday={showCurrentDateAsToday}
+                    showDateAsShortForm={showDateAsShortForm}
+                    onPreviousDayClick={onPreviousDayClick}
+                    onNextDayClick={onNextDayClick}
                     onRefresh={onRefresh}
+                    onCalendarDateSelect={onCalendarDateSelect}
                 />
-            </RowColumnHeader>
-            <RowHeaderColumn
-                $numOfRows={rowData.length}
-                $isScrolled={isScrolledX}
-                data-testid={"row-header-column-id"}
-            >
-                {renderRowHeaderColumn()}
-            </RowHeaderColumn>
-            <ColumnHeaderRow
-                $numOfColumns={hourlyIntervals.length}
-                $isScrolled={isScrolledY}
+                <EmptyTableContainer className="empty-container">
+                    {!loading ? (
+                        <NoResultsFound
+                            type="no-item-found"
+                            description={emptyContentMessage}
+                        />
+                    ) : (
+                        <Loader $isEmptyContent={isEmptyContent}></Loader>
+                    )}
+                </EmptyTableContainer>
+            </Container>
+        );
+    }
+
+    return (
+        <Container data-testid={testId} {...otherProps}>
+            <TimeTableHeader
+                data-id="timetable-header-id"
```

Comment:
realised the data-id goes to the DateNavigator component

```suggestion
                data-id="timetable-date-navigator"
```